Cython/Compiler/ModuleNode.py

```diff
@@ -401,6 +401,11 @@ def generate_cclass_include_code(self, type, i_code):
         i_code.dedent()
 
     def generate_c_code(self, env, options, result):
+        # Check for a common gotcha for new users: naming your .pyx file after the .c file you want to wrap
+        if not is_cython_generated_file(result.c_file, allow_failed=True):
```

Comment:
Is it worth covering `.h` files too? (and arguably `_api.h` files, although I suspect they're less likely to happen by accident)

Reply:
Yeah, I thought about that, too. Probably not a bad idea to give them a clear "Generated by Cython" header, is it?

Reply:
… except that every developer who still has a generated header file lying around would now be bothered to manually delete it, because we cannot currently detect that it was generated by Cython. :-/
I'll leave this out for now.

Reply:
When I look at it the generated headers (both .h and _api.h) they already start with `/* Generated by Cython <version tag> */`on both master and 0.29.x branches. I don't know how far back that goes, but I don't think it should be a problem.

Unless there's some way of generating headers that I'm missing (which is plausible too...)